infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationClientServerRouter.java

```diff
@@ -758,10 +757,25 @@ public void onConnectionDown(String nodeId, LogReplicationSession session) {
             }
         } catch (NullPointerException npe) {
             log.info("The replication components are already stopped for session {}", session);
+            return;
         }
 
-        if (isConnectionStarterForSession(session)) {
-            this.clientChannelAdapter.connectAsync(sessionToRemoteClusterDescriptor.get(session), nodeId, session);
+        try {
+            IRetry.build(IntervalRetry.class, () -> {
+                try {
+                    this.clientChannelAdapter.connectAsync(sessionToRemoteClusterDescriptor.get(session), nodeId, session);
+                } catch(NullPointerException npe) {
+                    log.info("The session is already stopped. Abort trying to reconnect {}", session);
```

Comment:
is there another way besides NullPointerException to determine that the session is already stopped?